packages/client/src/game/ui/HanabiCard.ts

```diff
@@ -570,15 +570,17 @@ export class HanabiCard extends Konva.Group implements NodeWithTooltip, UICard {
   }
 
   getBareNameRank(rankToShow: Rank | null): string {
-    if (this.isStackBase) {
-      return CARD_IMAGE_STACK_BASE_RANK_NAME;
+    // If a non-null rank is provided, it should take the highest precedence.
```

Comment:
what is the rank of a stack base? can we just delete the "if (stackbase)" part?

Reply:
`getBareNameRank` is used to compute the card's "bare name" which is used to render the correct image.  Something of the form `card-${bareNameSuit}-${bareNameRank}`.

So you might see `card-Teal-1` for the teal-1s in people's hands or if a teal 1 is on the stacks. If there's nothing on the stacks (the sort of "hollowed-out placeholder"), that has the name `card-Teal-${CARD_IMAGE_STACK_BASE_RANK_NAME}`.

The idea here is to allow users to (once again) override the stack-base name by having `rankToShow` apply first. But we _definitely_ still want to back off to CARD_IMAGE_STACK_BASE_RANK_NAME so that we show the stack placeholder in case there is no note or the stack is empty.

Reply:
then you should then copy that whole github comment into an actual code comment that fully explains what the code is doing

Reply:
happy to do that. (it's not like I wrote the code, btw... xD I just spent a while debugging it to figure out what it's doing. good idea to let future folks know what it does too...!)

will try to do this in the next commit.

Reply:
right, the idea is that we want to leave the code in a better state than when we started

Reply:
love it! (done)